File: core/Implementations/UserProcessor.java
package Implementations;

import Interfaces.AbstractUserProcessor;

/**
 * Created by billlai on 08/Oct/2016.
 */
class UserProcessor implements AbstractUserProcessor {
    public static final int SIGN_IN_SECCEEDED = 0;
    private static UserProcessor ourInstance = new UserProcessor();

    static UserProcessor getInstance() {
        return ourInstance;
    }

    private UserProcessor() {
    }

    @Override
    public boolean isUsernameAvailable(String username) {
        // TODO
        return false;
    }

    @Override
    public int signIn(String username, String password) {
        // TODO
        return 0;
    }

    @Override
    public int createUser(String username, String password) {
        // TODO
        return 0;
    }
}

File: src/iOS/UserProcessor.m
//
//  UserProcessor.m
//  Plus1
//
//  Created by ariya on 16/10/22.
//  Copyright © 2016年 ariya. All rights reserved.
//

#import <Foundation/Foundation.h>
#import "UserProcessor.h"



@implementation UserProcessor
    static UserProcessor *class = nil;
    + (UserProcessor*) getInstance{
        static UserProcessor *sharedInstance = nil;
        static dispatch_once_t onceToken;
        dispatch_once(&onceToken, ^(void) {
            sharedInstance = [[self alloc] initialize];
        });
        return sharedInstance;
    }
    
    - (id) initialize{
        if (self == [super init])
        {
            ;
        }
        return self;
    }
    
    - (bool) isUsernameAvailable:(NSString *)username{
        return true;
    }
    - (int) signIn:(NSString *)username :(NSString *)password{
        return 0;
    }
    - (int) createUser:(NSString *)username :(NSString *)password{
        return 0;
    }


@end
